src/zarr.cpp

```diff
@@ -587,6 +599,24 @@ zarr::Zarr::set_chunking(const ChunkingProps& props, const ChunkingMeta& meta)
     CHECK(planes_per_chunk_ > 0);
 }
 
+void
+zarr::Zarr::reset_state_()
```

Comment:
Nit: this is private, but `reset_state_` is not a great name given we have `Storage::state`, which this does not change.

Unfortunately, I don't have great ideas for different names. Maybe just `reset_` might be OK?

Reply:
Yeah, I didn't have any great ideas either. There's the "state" of it as a stateful object and there's the "device state" of it as a Storage device. Thinking of changing the comment in line 76 from `Setup` to `Setup / teardown`, so `teardown_` is an option. Again, not a great one.

Reply:
Another alternative is to remove this method and inline the code into `stop` since that's the only place it's called. Then we don't have to think of any names :p

Reply:
Fine by me.